
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Compass Settings &#8212; Compass 2021 documentation</title>
    
  <link rel="stylesheet" href="_static/css/index.73d71520a4ca3b99cfee5594769eaaae.css">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/basic.css" type="text/css" />
    
  <link rel="preload" as="script" href="_static/js/index.3da636dd464baa7582d2.js">

    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Compass Postprocessing" href="Compass-Postprocessing-Tutorial.html" />
    <link rel="prev" title="Tutorial" href="tutorial.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <nav class="navbar navbar-light navbar-expand-lg bg-light fixed-top bd-navbar" id="navbar-main">
<div class="container-xl">

    <a class="navbar-brand" href="index.html">
    
      <p class="title">Compass</p>
    
    </a>
    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbar-menu" aria-controls="navbar-menu" aria-expanded="false" aria-label="Toggle navigation">
        <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar-menu" class="col-lg-9 collapse navbar-collapse">
      <ul id="navbar-main-elements" class="navbar-nav mr-auto">
        
        
        <li class="nav-item ">
            <a class="nav-link" href="install.html">Installing Compass</a>
        </li>
        
        <li class="nav-item active">
            <a class="nav-link" href="user-guide.html">Compass User Guide</a>
        </li>
        
        <li class="nav-item ">
            <a class="nav-link" href="Compass-Postprocessing-Tutorial.html">Compass Postprocessing</a>
        </li>
        
        <li class="nav-item active">
            <a class="nav-link" href="">Compass Settings</a>
        </li>
        
        
      </ul>


      

      <ul class="navbar-nav">
        
        
      </ul>
    </div>
</div>
    </nav>
    

    <div class="container-xl">
      <div class="row">
          
          <div class="col-12 col-md-3 bd-sidebar"><form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">

    <div class="bd-toc-item active">
    
  
    <ul class="nav bd-sidenav">
        
        
        
        
          
            
                <li class="">
                    <a href="install.html">Installing Compass</a>
                </li>
            
          
            
                <li class="">
                    <a href="tutorial.html">Tutorial</a>
                </li>
            
          
            
                <li class="active">
                    <a href="">Compass Settings</a>
                </li>
            
          
            
                <li class="">
                    <a href="Compass-Postprocessing-Tutorial.html">Compass Postprocessing</a>
                </li>
            
          
            
                <li class="">
                    <a href="AWS-tutorial.html">AWS Tutorial</a>
                </li>
            
          
        
        
        
        
        
          
        
        
      </ul>
  
  </nav>
          </div>
          

          
          <div class="d-none d-xl-block col-xl-2 bd-toc">
              
<div class="tocsection onthispage pt-5 pb-3">
    <i class="fas fa-list"></i> On this page
</div>

<nav id="bd-toc-nav">
    <ul class="nav section-nav flex-column">
    
        <li class="nav-item toc-entry toc-h2">
            <a href="#input-and-output-settings" class="nav-link">Input and Output settings</a>
        </li>
    
        <li class="nav-item toc-entry toc-h2">
            <a href="#metabolic-model-settings" class="nav-link">Metabolic Model Settings</a>
        </li>
    
        <li class="nav-item toc-entry toc-h2">
            <a href="#penalty-settings" class="nav-link">Penalty Settings</a>
        </li>
    
        <li class="nav-item toc-entry toc-h2">
            <a href="#computing-settings" class="nav-link">Computing Settings</a>
        </li>
    
        <li class="nav-item toc-entry toc-h2">
            <a href="#testing-and-hidden-settings" class="nav-link">Testing and Hidden Settings</a>
        </li>
    
    </ul>
</nav>


              
          </div>
          

          
          <main class="col-12 col-md-9 col-xl-7 py-md-5 pl-md-5 pr-md-4 bd-content" role="main">
              
              <div>
                
  <div class="section" id="compass-settings">
<h1>Compass Settings<a class="headerlink" href="#compass-settings" title="Permalink to this headline">¶</a></h1>
<div class="contents local topic" id="contents">
<p class="topic-title">Contents</p>
<ul class="simple">
<li><p><a class="reference internal" href="#input-and-output-settings" id="id1">Input and Output settings</a></p></li>
<li><p><a class="reference internal" href="#metabolic-model-settings" id="id2">Metabolic Model Settings</a></p></li>
<li><p><a class="reference internal" href="#penalty-settings" id="id3">Penalty Settings</a></p></li>
<li><p><a class="reference internal" href="#computing-settings" id="id4">Computing Settings</a></p></li>
<li><p><a class="reference internal" href="#testing-and-hidden-settings" id="id5">Testing and Hidden Settings</a></p></li>
</ul>
</div>
<p>Compass allows users to customize various features:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>usage: Compass <span class="o">[</span>-h<span class="o">]</span> <span class="o">[</span>--data FILES <span class="o">[</span>FILES ...<span class="o">]]</span> <span class="o">[</span>--model MODEL<span class="o">]</span>
            <span class="o">[</span>--species SPECIES<span class="o">]</span> <span class="o">[</span>--media MEDIA<span class="o">]</span> <span class="o">[</span>--output-dir DIR<span class="o">]</span>
            <span class="o">[</span>--temp-dir DIR<span class="o">]</span> <span class="o">[</span>--torque-queue QUEUE<span class="o">]</span> <span class="o">[</span>--num-processes N<span class="o">]</span>
            <span class="o">[</span>--lambda F<span class="o">]</span> <span class="o">[</span>--num-threads N<span class="o">]</span> <span class="o">[</span>--and-function FXN<span class="o">]</span>
            <span class="o">[</span>--select_reactions FILE<span class="o">]</span> <span class="o">[</span>--num-neighbors N<span class="o">]</span>
            <span class="o">[</span>--symmetric-kernel<span class="o">]</span> <span class="o">[</span>--input-weights FILE<span class="o">]</span>
            <span class="o">[</span>--penalty-diffusion MODE<span class="o">]</span> <span class="o">[</span>--no-reactions<span class="o">]</span>
            <span class="o">[</span>--calc-metabolites<span class="o">]</span> <span class="o">[</span>--precache<span class="o">]</span> <span class="o">[</span>--input-knn FILE<span class="o">]</span>
            <span class="o">[</span>--output-knn FILE<span class="o">]</span> <span class="o">[</span>--latent-space FILE<span class="o">]</span> <span class="o">[</span>--list-genes FILE<span class="o">]</span>
</pre></div>
</div>
<p>Below we describe the features in more detail:</p>
<div class="section" id="input-and-output-settings">
<h2><a class="toc-backref" href="#id1">Input and Output settings</a><a class="headerlink" href="#input-and-output-settings" title="Permalink to this headline">¶</a></h2>
<dl>
<dt><strong>--data</strong> [FILES]</dt><dd><p>File to input gene expression data with rows as genes and columns as sample names.
The input can a single tab-delimited file with row and column labels:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>--data expression.tsv
</pre></div>
</div>
<p>Or the input can be in the matrix market format followed by two tab separted files, with the first file being row labels corresponding to genes and the second file being column labels corresponding to sample names:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>--data expression.mtx genes.tsv sample_names.tsv
</pre></div>
</div>
<p>The sample names file can be omitted in which case the samples will be labelled by index, but the genes file must be included.</p>
</dd>
<dt><strong>--output-dir</strong> [DIR]</dt><dd><p>Final directory to output concatenated reactions.txt file</p>
</dd>
<dt><strong>--temp-dir</strong> [DIR]</dt><dd><p>Directory to store partial results for completed
samples in a dataset</p>
</dd>
</dl>
</div>
<div class="section" id="metabolic-model-settings">
<h2><a class="toc-backref" href="#id2">Metabolic Model Settings</a><a class="headerlink" href="#metabolic-model-settings" title="Permalink to this headline">¶</a></h2>
<dl class="simple">
<dt><strong>--model</strong> [MODEL]</dt><dd><p>Metabolic model to use. Options:</p>
<ul class="simple">
<li><p>RECON1_mat</p></li>
<li><p>RECON2_mat (default)</p></li>
<li><p>RECON2.2</p></li>
</ul>
</dd>
<dt><strong>--media</strong> [MEDIA]</dt><dd><p>The media to simulate the model with.</p>
</dd>
<dt><strong>--species</strong> [SPECIES]</dt><dd><p>Species to use to match genes to model. Options:</p>
<ul class="simple">
<li><p>homo_sapiens (default)</p></li>
<li><p>mus_musculus</p></li>
</ul>
</dd>
<dt><strong>--and-function</strong> [FXN]</dt><dd><p>Which function used to aggregate and
associations. Options:</p>
<ul class="simple">
<li><p>min</p></li>
<li><p>median</p></li>
<li><p>mean (default)</p></li>
</ul>
</dd>
<dt><strong>--calc-metabolites</strong></dt><dd><p>Flag to enable calculation and output of
uptake/secretion scores in addition to reaction scores.</p>
</dd>
<dt><strong>--no-reactions</strong></dt><dd><p>Flag to disable calculation and output of reaction
scores in addition to uptake/secretion scores.</p>
</dd>
<dt><strong>--list-genes</strong> [FILE]</dt><dd><p>File to output a list of metabolic genes needed for selected metabolic model.</p>
</dd>
<dt><strong>--select-reactions</strong> [FILE]</dt><dd><p>Compute compass scores only for the reactions listed in the given file.
FILE is expected to be textual, with one line per reaction
(undirected, namely adding the suffix &quot;_pos&quot; or &quot;_neg&quot; to a line will create a valid directed reaction id).
Unrecognized reactions in FILE are ignored.</p>
</dd>
</dl>
</div>
<div class="section" id="penalty-settings">
<h2><a class="toc-backref" href="#id3">Penalty Settings</a><a class="headerlink" href="#penalty-settings" title="Permalink to this headline">¶</a></h2>
<dl class="simple">
<dt><strong>--penalty-diffusion</strong> [MODE]</dt><dd><p>Mode to use to share reaction penalty values
between single cells. Options:</p>
<ul class="simple">
<li><p>gaussian (default)</p></li>
<li><p>knn</p></li>
</ul>
</dd>
<dt><strong>--lambda</strong> [F]</dt><dd><p>Smoothing factor for single-cell data. Should be set between 0 and 1</p>
</dd>
<dt><strong>--num-neighbors</strong> [K]</dt><dd><p>Either effective number of neighbors for gaussian
penalty diffusion or exact number of neighbors for KNN penalty
diffusion. Default is 30</p>
</dd>
<dt><strong>--input-weights</strong> [FILE]</dt><dd><p>File to input custom weights for averaging of single-cell data.
The column and row labels should be the same as the names of samples in expression data.</p>
</dd>
<dt><strong>--symmetric-kernel</strong></dt><dd><p>Flag to enable symmetrizing the TSNE kernel which takes longer</p>
</dd>
<dt><strong>--input-knn</strong> [FILE]</dt><dd><p>File to input a precomputed kNN graph for the samples.
File must be a tsv with one row per sample and (k+1) columns.
The first column should be sample names, and the next k columns should be indices of the k nearest neighbors (by their order in column 1).</p>
</dd>
<dt><strong>--output-knn</strong> [FILE]</dt><dd><p>File to save kNN graph of the samples to.
File will be a tsv with one row per sample and (k+1) columns.
The first column will be sample names, and the next k columns will be indices of the k nearest neighbors (by their order in column 1).</p>
</dd>
<dt><strong>--latent-space</strong> [FILE]</dt><dd><p>File with latent space representation of the samples on which to do the kNN clustering.
Should be a tsv with one row per sample and one column per dimension of the latent space.</p>
</dd>
</dl>
</div>
<div class="section" id="computing-settings">
<h2><a class="toc-backref" href="#id4">Computing Settings</a><a class="headerlink" href="#computing-settings" title="Permalink to this headline">¶</a></h2>
<dl class="simple">
<dt><strong>--num-processes</strong> [N]</dt><dd><p>Number of processes for Compass to use. Must be a positive integer and defaults to the number of processors on machine (using Python’s <code class="code docutils literal notranslate"><span class="pre">multiprocessing.cpu_count()</span></code>). Ignored
when submitting job onto a queue</p>
</dd>
<dt><strong>--num-threads</strong> [N]</dt><dd><p>Number of threads to use per sample. Default is 1.</p>
</dd>
</dl>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>It is generally better to increase the number of processes than the number of threads for better performance, unless the number of processes is greater than the number of samples.</p>
</div>
<dl class="simple">
<dt><strong>--torque-queue</strong> [QUEUE]</dt><dd><p>Name of the torque queue to submit to</p>
</dd>
<dt><strong>--precache</strong></dt><dd><p>A flag to force compass to build up the cache for the input selected model and media. This will rebuild the cache even if one already exists.</p>
</dd>
</dl>
</div>
<div class="section" id="testing-and-hidden-settings">
<h2><a class="toc-backref" href="#id5">Testing and Hidden Settings</a><a class="headerlink" href="#testing-and-hidden-settings" title="Permalink to this headline">¶</a></h2>
<p>There are several Compass arguments which are not listed by the parser because they are primarily for testing or for batch jobs.</p>
<dl class="simple">
<dt><strong>--test-mode</strong></dt><dd><p>Flag which limits computing scores to the first 100 reactions and first 50 metabolites</p>
</dd>
<dt><strong>--detailed-perf</strong></dt><dd><p>Flag which enables more performance data collection such as runtimes per reaction per sample.</p>
</dd>
<dt><strong>--collect</strong></dt><dd><p>Flag to have compass collect results. Used for batch jobs</p>
</dd>
<dt><strong>--config-file</strong> [FILE]</dt><dd><p>Setting used for batch jobs</p>
</dd>
<dt><strong>--penalties-file</strong> [FILE]</dt><dd><p>File which allows for specifying a penalties file other than the default one (which is _tmp/penalties.txt.gz)</p>
</dd>
<dt><strong>--lpmethod</strong> [N]</dt><dd><p>Argument to choose the algorithm CPLEX uses.
See <a class="reference external" href="https://www.ibm.com/support/knowledgecenter/SSSA5P_20.1.0/ilog.odms.cplex.help/CPLEX/Parameters/topics/LPMETHOD.html">Cplex documentation for more details</a>.
Through testing the barrier algorithm (4) is fastest and therefore default, with automatic selection (0) or dual simplex (2) also performing well.</p>
</dd>
<dt><strong>--advance</strong> [N]</dt><dd><p>Argument to choose the setting for Cplex’s advanced basis setting.
See <a class="reference external" href="https://www.ibm.com/support/knowledgecenter/SSSA5P_20.1.0/ilog.odms.cplex.help/CPLEX/Parameters/topics/AdvInd.html">Cplex documentaton for more details</a>.
Defaults to 2 as best runtime was found using that for tests.</p>
</dd>
<dt><strong>--save-argmaxes</strong></dt><dd><p>Flag to enable saving the argmaxes for computing Compass scores of each reaction. Fun fact: solving the TSP greedily on the argmaxes graph to make full use of the advanced basis setting with the simplex algorithm did not outperform the barrier algorithm in practice.</p>
</dd>
</dl>
</div>
</div>


              </div>
              
              
              <div class='prev-next-bottom'>
                
    <a class='left-prev' id="prev-link" href="tutorial.html" title="previous page">Tutorial</a>
    <a class='right-next' id="next-link" href="Compass-Postprocessing-Tutorial.html" title="next page">Compass Postprocessing</a>

              </div>
              
          </main>
          

      </div>
    </div>

    
  <script src="_static/js/index.3da636dd464baa7582d2.js"></script>


    <footer class="footer mt-5 mt-md-0">
  <div class="container">
    <p>
          &copy; Copyright 2021, Yosef Lab.<br/>
        Created using <a href="http://sphinx-doc.org/">Sphinx</a> 3.4.1.<br/>
    </p>
  </div>
</footer>
  </body>
</html>